Add docs for Next.js SDK option `autoInstrumentServerFunctions` (#5542)
getsentry/sentry-docs: src/platforms/javascript/guides/nextjs/manual-setup.mdx

In that case you can also skip the `sentry-cli` configuration step below.

> ### Automatically Instrument API Routes And Data Fetching Methods

Reviewer: ```suggestion
### Automatically Instrument API Routes and Data Fetching Methods
```

_(New in version TODO)_

> The SDK provides an option to automatically instrument API routes and serverside [Next.js Data Fetching Methods](https://nextjs.org/docs/basic-features/data-fetching/overview) with error and performance monitoring, removing the need to manually wrap API routes in `withSentry`.

Reviewer: ```suggestion
The SDK provides an option to automatically instrument API routes and server-side [Next.js data fetching methods](https://nextjs.org/docs/basic-features/data-fetching/overview) with error and performance monitoring, removing the need to manually wrap API routes in `withSentry`:
```

```javascript {filename:next.config.js}
const moduleExports = {
  sentry: {
    autoInstrumentServerFunctions: true,
  },
};
```

Under the hood, the SDK is using a Webpack loader to wrap all your API route handlers and data fetching methods.

> In case the automatic instrumentation does not work for your use case, API routes can also be wrapped manually using the `withSentry` function.

Reviewer: ```suggestion
If the automatic instrumentation doesn't work for your use case, API routes can also be wrapped manually using the `withSentry` function:
```

```javascript {filename:pages/api/*}
import { withSentry } from "@sentry/nextjs";

const handler = (req, res) => {
  res.status(200).json({ name: "John Doe" });
};

export default withSentry(handler);
```

```typescript {filename:pages/api/*}
import type { NextApiRequest, NextApiResponse } from "next"
import { withSentry } from "@sentry/nextjs";

const handler = (req: NextApiRequest, res: NextApiResponse) => {
  res.status(200).json({ name: "John Doe" });
};

export default withSentry(handler);
```

> Data Fetching Methods can also be manually wrapped using the following functions:

Reviewer: ```suggestion
Data fetching methods can also be manually wrapped using the following functions:
```
